packages/wrangler/src/config/validation.ts

```diff
@@ -2773,6 +2773,12 @@ const validateBindingsHaveUniqueNames = (
 				bindingsGroupedByName[bindingName] = [];
 			}
 
+			if (bindingName.toUpperCase() === "ASSETS" && isPagesConfig(config)) {
+				diagnostics.errors.push(
+					`The name 'ASSETS' is reserved in Pages projects. Please use a different name for your ${bindingType} binding.`
```

Comment:
Nit: Is binding name is case sensitive? Might be better to show the original name user provided.
```suggestion
					`The name '${bindingName}' is reserved in Pages projects. Please use a different name for your ${bindingType} binding.`
```

Reply:
turns out it is case sensitive, so you totally can deploy with 'assets' (lowercase) no issues